lib/grppccom.gi

```diff
@@ -1207,15 +1207,51 @@ end);
 #M  ComplementcClassesRepresentatives( <G>, <N> )
 ##
 InstallMethod( ComplementClassesRepresentatives,
-  "tell that the normal subgroup must be solvable",IsIdenticalObj,
-  [IsGroup,IsGroup],-2*RankFilter(IsGroup),
+  "tell that the normal subgroup or factor must be solvable", IsIdenticalObj,
+  [ IsGroup, IsGroup ], -2*RankFilter(IsGroup),
 function( G, N )
-  if IsSolvableGroup(N) then
+  if IsSolvableGroup(N) or HasSolvableFactorGroup(G, N) then
     TryNextMethod();
   fi;
-  Error("Cannot compute complement classes for nonsolvable normal subgroups");
+  Error("Cannot compute complements if both N and G/N are nonsolvable");
 end);
```

Comment:
I added tests for the errors and for the G=N case. However, this ````TryNextMethod```` should never trigger, because this whole method only triggers if the main method exits with ````TryNextMethod````, that is if both N and G/N are nonsolvable.